gcp/observability/config.go

```diff
@@ -59,6 +54,28 @@ func fetchDefaultProjectID(ctx context.Context) string {
 	return credentials.ProjectID
 }
 
+// validateMethodString validates whether the string passed in is a valid
+// pattern.
+func validateMethodString(method string) error {
+	if method == "*" {
```

Comment:
Note that this function isn't even called if `method` is `*`.

Maybe inline this in `validateLogEventMethod`?

Reply:
Yeah fair point missed that. I'm assuming inline this means just leave what is there (if * continue (skip calling this function/validating). Deleted this conditional.